### PR TITLE
[V3] Docker Compose 

### PR DESCRIPTION
**Pull request by mikeshardmind**
### Type

- [ ] Bugfix
- [ ] Enhancement
- [X] New feature

### Description of the changes

Docker compose solution with persistent storage through volumes and safe handling of tokens with secrets. Documentation not here yet. Needs a token in a file at
`docker/settings/TOKEN` 
and a prefix at 
`docker/settings/PREFIX`

### CODE REVIEW

File: docker/settings/TOKEN.example [hunk 1]
```diff
@@ -0,0 +1 @@
+notactuallyatokenyoushallnotpassblargh
```
Comment (tekulvw) [inline docker/settings/TOKEN.example:1]: Why can't this go into `docker/.env`?

Comment (mikeshardmind) [reply docker/settings/TOKEN.example:1]: becuase if we deploy an image, anything in env will show up in inspect iimage (including any tokens used for testing.

**Comment by mikeshardmind** [comment]
Switched the base image from python:3.6-jessie to alpine:3.7 , made other modifications to match that resulting in a drop in image size from ~960MB to ~170MB

**Comment by mikeshardmind** [comment]
Still more to do here including stronger documentation, and possible changes pending feedback.
Current testing steps for running without docker-compose
```
docker build . -t redbot
docker run --env-file .env -e UID=$(id -u) -v `pwd`/data:/data -t -d redbot
```
contents of .env file (template)
```
RED_TOKEN=token here
PREFIX=prefix here
```
compose solution still works with this.

**Comment by mikeshardmind** [comment]
This needs enough changes that I'm closing this. Will re-open when the details have been discussed more in the discord, and a solution fitting that has been made.

**Pull request closed without merging**